make/e2e-setup.mk

```diff
@@ -36,6 +36,10 @@ IMAGE_kind_arm64 := $(IMAGE_kind_amd64)
 PEBBLE_COMMIT = ba5f81dd80fa870cbc19326f2d5a46f45f0b5ee3
 GATEWAY_API_VERSION = 0.4.1
 
+# TODO: move the installation commands in this file to separate scripts like those in https://github.com/cert-manager/cert-manager/tree/master/devel/addon for readability
+# Once that is done, we can consume this variable from ./make/config/lib.sh
+SERVICE_IP_PREFIX = 10.0.0
```

Comment:
I agree with this approach to the problem. We can do this better later for sure!

I've said before that I increasingly think our e2e setup should be implemented as a finite state machine, and I think every time I look at `make/e2e-setup.mk` that's reinforced for me. In any case, this problem space doesn't neatly map onto make targets because we're not dealing with files. I think we want several PHONY targets as frontends for backend scripts (which I reckon might be worth doing in a "real" language like go rather than bash)

All of this is irrelevant for this PR, I'm just stood on my soapbox here :grin:

Reply:
>  I think we want several PHONY targets as frontends for backend scripts (which I reckon might be worth doing in a "real" language like go rather than bash)

That sounds like a great approach to me
Yeah I think it makes sense for targets that install stuff with Helm to be phony. I think Helm already does the work of determining if there are changes needed so there's no need to use make for that. (I wonder if similar wouldn't be the case with other targets that make images for example)